lib/wiki_api.rb

```diff
@@ -51,51 +51,56 @@ def get_article_rating(titles)
-    raw.map do |_article_id, talkpage|
-      # Remove "Talk:" from the "title" value to get the title.
-      { talkpage['title'][5..-1].tr(' ', '_') =>
-        parse_article_rating(talkpage) }
-    end
-  end
-
-  ###################
-  # Parsing methods #
-  ###################
-
-  def parse_article_rating(raw_talk)
-    # Handle MediaWiki API errors
-    return nil if raw_talk.nil?
-    # Handle the case of nonexistent talk pages.
-    return nil if raw_talk['missing']
-
-    wikitext = raw_talk['revisions'][0]['*']
-    ArticleClassExtractor.new(wikitext).extract
+    query_parameters = { titles: titles,
+                         prop: 'pageassessments', 
+                         formatversion: '2',
+                         redirects: 'true' }
+    response = get_responses_continue(query_parameters)
+    return [] unless response
+    return [] unless response['query'] 
+
+    info = response['query']['pages']
+    return [] unless info
+    ArticleClassFetcher.new(info).get_article_rating 
   end
 
   #####################
   # Other API methods #
   #####################
 
-  # Get raw page content for one or more pages titles, which can be parsed to
-  # find the article ratings. (The corresponding Talk page are the one with the
-  # relevant info.) Example query:
-  # http://en.wikipedia.org/w/api.php?action=query&prop=revisions&rvprop=content&rawcontinue=true&redirects=true&titles=Talk:Selfie
-  def get_raw_page_content(article_titles)
-    query_parameters = { titles: article_titles,
-                         prop: 'revisions',
-                         rvprop: 'content' }
-    info = mediawiki('query', query_parameters)
-    return if info.nil?
-    page = info.data['pages']
-    page.nil? ? nil : page
+  def get_responses_continue(query_parameters)
+    # Fetches response of pages till batchcomplete=true with formatversion=2 to get a complete response that can be processed together
+    # Rawcontinue should not be true 
+    # Query parameter should contain an array of 'pages'
+    # https://en.wikipedia.org/w/api.php/?action=query&prop=pageassessments&formatversion=2&titles=Apple|Pear
+    query_parameters['formatversion'] = '2'
+    response = nil
+
+    loop do 
+      new_response = mediawiki('query', query_parameters)
+      break if new_response.nil? 
+      if response.nil? 
+        response = new_response
+      elsif response['query'] != nil && new_response['query'] != nil
+          values = new_response['query']['pages']
+          if response['query']['pages'] != nil
+            new_response['query']['pages'] = response['query']['pages'].concat values 
+          else
+            response['query']['pages'] = values
+          end
+          response = new_response
+      end
+      if new_response['continue'] != nil
+        new_response['continue'].map do |mod, parameter|
+          query_parameters[mod] = parameter 
+        end
+      else
+        break
+      end
+    end
```

Comment:
This is a very complex method, especially with the nesting and the if/else blocks. I suspect this can be simplified considerably. The `continue` implementation in `lib/commons.rb` may be a useful point of reference.